python/paddle/fluid/contrib/slim/quantization/quantization_mkldnn_pass.py

```diff
@@ -254,3 +274,276 @@ def _remove_unused_var_nodes(self, graph):
                             graph.all_var_nodes())
         }
         graph.safe_remove_nodes(all_unused_vars)
+
+
+class FakeQAT2MkldnnINT8PerfPass(object):
+    """
+    Transform a QAT model IrGraph into MKL-DNN supported INT8 IrGraph.
+    The pass consists of the following transformations:
+        1. gather scale values from fake quantize/dequantize operators,
+        2. extract FP32 inference model graph from the QAT graph, i.e.
+            a.  remove fake quantize/dequantize operators,
+            b.  dequantize conv2d and mul's weights,
+        3. optimize the FP32 graph using standard FP32 optimization fuses
+            (e.g. `conv2d`+`bn` -> `conv2d`),
+        4. quantize the optimized FP32 graph using standard INT8v2 quantization
+            passes (`cpu_quantize_pass`, `cpu_quantize_squash_pass`).
+    """
+
+    def __init__(self, _scope=None, _place=None, _core=None, _debug=False):
+        self._scope = _scope
+        self._place = _place
+        self._core = _core
+        self._debug = _debug
+        self._quantize_types = [
+            'fake_quantize_moving_average_abs_max',
+            'fake_quantize_range_abs_max',
+            'fake_quantize_dequantize_moving_average_abs_max'
+        ]
+        self._fake_quantize_types = [
+            'fake_quantize_moving_average_abs_max',
+            'fake_quantize_dequantize_moving_average_abs_max'
+        ]
+        self._fake_dequantize_types = ['fake_dequantize_max_abs']
+        self._conv_ops = ['conv2d', 'depthwise_conv2d']
+        self._pool_ops = ['pool2d']
+        self._mul_ops = ['mul']
+        self._fc_ops = ['fc']
+        self._weight_scales = {}
+        # Collect the Input and Output sclaes from Fake QAT models
```

Comment:
For example, in this statement `self._var_quant_scales[input_name] = (False, lod_tensor)` , what is the False for? How about True?

Reply:
VarQuantScales hold a map which key of are tensor names and values are pairs of (bool, Tensor).
The bool tells us **whether the data in that tensor can be quantized to uint8** and tensor contains a scale corresponding to the tensor.

Since QAT has been using int8 tensors everywhere where applicable, we decided to set this bool to False, so that no tensor is quantized to use uint8 datatype.